plugin.rb

```diff
@@ -567,12 +567,15 @@ def self.denormalized_columns
 
   # Reimplement a couple ActiveRecord methods, but use PluginStore for storage instead
   class DataExplorer::Query
-    attr_accessor :id, :name, :description, :sql
+    attr_accessor :id, :name, :description, :sql, :user, :time
 
     def initialize
       @name = 'Unnamed Query'
       @description = 'Enter a description here'
       @sql = 'SELECT 1'
+      #TODO: Figure out where to assign current user for storage
+      @user = 'Admin'
```

Comment:
@techAPJ 
I need to send the current user's name through the Serializer. 
Where would be a good place to assign it? `current_user.username` does not seem to work inside initialize()

Reply:
Instead of initializing query with user and time, I think we should pass them on create via controller.

Reply:
@techAPJ, so I should remove all occurrences of user and time from this file and write a new component which will set time and user? Any relevant example for this?

Reply:
@techAPJ, so I should remove all occurrences of user and time from this file and write a new component which will set time and user? 
Or do you only want me to remove user and time from initialize()?

Reply:
Serializer is used to pass data to the client (front-end). In this case (saving user and timestamp in a row) everything can happen on server (back-end) because server already knows the current (logged in) user. What you need to do is that pass the current user and timestamp when the (rails) controller receives create request [here](https://github.com/discourse/discourse-data-explorer/blob/master/plugin.rb#L937-L945).

No need to initialize the empty query with user_id and timestamp. Note that how `id` is allocated on runtime. Also instead of `username` we should save user `id` in `created_by` field.

Reply:
Putting the definition in create makes so much sense, understood!
I've fixed that, thank you!